### PR TITLE
CCache on windows

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR ensures ccache is configured on windows when building with Ninja. It should hopefully mean far quicker build speeds when modifying certain files.

**To test:**
1. Update your environment to make sure ccache is installed. You can do `mamba env update -f mantid-developer-win.yml --prune`
1. Starting with an empty build folder, create a ninja build by first opening `x64 Native Tools command prompt for VS 2019`, activating the `mantid-developer` conda env and running `cmake --preset=win-ninja` from the source directory, and `ninja` from the build directory.
2. Change a commonly used header file like `Instrument.h`
3. Do ninja again in the build directory, 
4. Change `Instrument.h` back to its preview form. Do ninja again from the build directory and it should build fairly quickly
4. Check your `AppData/Roaming` directory for a `ccache` folder, to check that caching is being done
5. Create an MSVC build with `cmake --preset=win-msvc -B=buildmsvc`. Build it using the visual studio IDE and check that debugging still works (i.e. you can still git breakpoints)

Part of #34490

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DanielMurphy22 — DISMISSED** [review]
Very good! Makes a real difference to rebuild time!

File: buildconfig/CMake/WindowsSetup.cmake [hunk 1]
```diff
@@ -59,10 +59,16 @@ if(CONDA_ENV)
   # switch off all optimzations
   set(_conda_debug_cfg_name DebugWithRelRuntime)
   string(TOUPPER ${_conda_debug_cfg_name} _conda_debug_cfg_name_upper)
+  # Set language flags for Visual studio and ninja generators
+  if(CCACHE_FOUND)
+    set(_language_flags "/Z7 /Ob0 /Od /RTC1")
+  else()
+    set(_language_flags "/Zi /Ob0 /Od /RTC1")
+  endif()
```
Comment (DanielMurphy22) [inline buildconfig/CMake/WindowsSetup.cmake:67]:
```suggestion
  # for a single-config generator like Ninja, aim to use CCache and Debug on Clion. The Zi flag is not supported by 
  CCache so we need to use Z7 to allow debugging.
  if(CCACHE_FOUND)
    set(_language_flags "/Z7 /Ob0 /Od /RTC1")
  else()  # For multi-config generators like MSVC, Ccache is not used, so Zi flag can be used for debugging
    set(_language_flags "/Zi /Ob0 /Od /RTC1")
  endif()
```

Feel free to adjust!

**Review by DanielMurphy22 — DISMISSED** [review]
Much happier!

File: buildconfig/CMake/WindowsSetup.cmake [hunk 2]
```diff
@@ -59,10 +59,18 @@ if(CONDA_ENV)
   # switch off all optimzations
   set(_conda_debug_cfg_name DebugWithRelRuntime)
   string(TOUPPER ${_conda_debug_cfg_name} _conda_debug_cfg_name_upper)
+  if(CCACHE_FOUND)
```
Comment (martyngigg) [inline buildconfig/CMake/WindowsSetup.cmake:62]: As we have the `USE_CCACHE` option too I think we need to check that?

```suggestion
  if(USE_CCACHE AND CCACHE_FOUND)
```

as if cmake was run first with `USE_CCACHE=ON` but then run again with ``USE_CCACHE=OFF` ccache was still found but the developer no longer wants it to be used.

**Review by martyngigg — CHANGES_REQUESTED** [review]
This looks great.

Just one minor thing with an `if` and I think it's good to go.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**